extension/source/Home/Wallet/Wallets/Recovery/StepThreeRecover.tsx

```diff
@@ -0,0 +1,92 @@
+import { ethers } from 'ethers';
+import { Download } from 'phosphor-react';
+import { FunctionComponent, useState } from 'react';
+import Button from '../../../../components/Button';
+import { useQuill } from '../../../../QuillContext';
+
+const StepThreeRecover: FunctionComponent<{
+  onComplete: () => void;
+  walletPrivateKey: string;
+}> = ({ onComplete, walletPrivateKey }) => {
+  const { rpc } = useQuill();
+
+  const [salt, setSalt] = useState<string>('');
+  const [instantWalletAddress, setInstantWalletAddress] = useState<string>('');
+
+  const handleRecover = async () => {
+    await rpc.addRecoveryWallet(
+      instantWalletAddress,
+      ethers.utils.formatBytes32String(salt),
+      walletPrivateKey,
+    );
```

Comment:
Do we need an error state/handling if this RPC call fails?

Reply:
error-handling is not great across the extension's codebase. I think we should have a separate task for improving errors/fails altogether. That's why left it simple here as well.